interface/main/finder/dynamic_finder.php

```diff
@@ -398,7 +398,62 @@ function rp()
     }
     $patientService = new PatientService();
     $rp = $patientService->getRecentPatientList();
-    return ['headers' => $headers, 'rp' => $rp];
+    // Get a list of the columns in patient_data that are either date or datetime:
+    $sql_dtCols = "select column_name, data_type from information_schema.columns where table_schema = DATABASE() and TABLE_NAME = 'patient_data' and (data_type = 'datetime' OR data_type = 'date')";
+    $res_dtCols = sqlStatement($sql_dtCols);
+    $pd_dtCols = [];
+    while ($row = sqlFetchArray($res_dtCols)) {
+        $pd_dtCols[] = $row;
+    }
+    $date_cols = [];
+    $datetime_cols = [];
+    foreach ($pd_dtCols as $k => $v) {
+        if ($v['data_type'] == "datetime") {
+            $datetime_cols[] = $v['column_name'];
+        } else if ($v['data_type'] == "date") {
+            $date_cols[] = $v['column_name'];
+        }
+    }
+    // Build SQL statement to pull desired columns from patient_data table...
+    $pd_sql = "SELECT id as pid";
```

Comment:
can't consider `id` and `pid` to be interchangeable

Reply:
Ah,  there isn't a PRIMARY KEY on patient_data.  Would it be better to use id or pid as the way of identifying the patients we want to have in the recent patient list?

Reply:
`pid`

Reply:
I've updated my code to use pid.